frontend/src/Context/BackgroundPublisherProvider/useBackgroundPublisher/useBackgroundPublisher.tsx

```
@@ -222,6 +234,79 @@ const useBackgroundPublisher = (): BackgroundPublisherContextType => {
     setIsVideoEnabled(!isVideoEnabled);
   };
 
+  /**
+   * Add a custom background image
+   * @param {string} dataUrl - The data URL of the image to add
+   * @returns {void}
+   */
+  const addCustomImage = useCallback(
+    (dataUrl: string) => {
+      addImageToStorage(dataUrl);
+      setCustomImages(getImagesFromStorage());
+    },
+    [getImagesFromStorage, addImageToStorage]
+  );
+
+  /**
+   * Delete a custom background image
+   * If the deleted image is currently selected, clear the background filter
+   * @param {string} id - The ID of the image to delete
+   * @returns {void}
+   */
+  const deleteCustomImage = useCallback(
+    (id: string) => {
+      const imageToDelete = customImages.find((img) => img.id === id);
+      if (!imageToDelete) {
+        return;
+      }
+
+      // Don't allow deletion if this image is currently selected
+      if (backgroundSelected === imageToDelete.dataUrl) {
+        return;
```

Hi @OscarFava , thanks a lot for the changes ! A couple of things:

For these two validations `if (!imageToDelete) {`  and `backgroundSelected === imageToDelete.dataUrl `, let's throw an exception instead of just returning.

The function is call `deleteCustomImage`, if I try to delete something that doesn't exists or something I should not be deleting I should get an exception probably

also a **nic** but we really want to enforce named conditions so: `backgroundSelected === imageToDelete.dataUrl` that condition let's use a named condition

```tsx
const isSelectedBackground = backgroundSelected === imageToDelete.dataUrl
if(isSelectedBackground)
```

Finally is it **currentBackgroundFilter** the the same than then **backgroundSelected**?  If so, currently its impossible to deleted the image if on use so maybe we don't need this piece of code here.